File: Factor/Python/Enemy.py
# -*- coding: utf-8 -*-
"""
Created on Sat May  7 18:04:32 2022

@author: MahmoudSaeed
"""
from abc import ABC

class Enemy():
    # def __init__(self , name , damage , health):
    #     self.name = name
    #     self.damage = damage
    #     self.health = health
        
        
    def setName(self , name):
        print(name,"_2")
        self.name = name
    
    def setDamage(self , damage):
        self.damage = damage
        
    def setHealth(self , health):
        self.health = health
    
    # def getHealth(self):
    #     return self.health
    
    # def getName(self):
    #     return self.name
    
    # def getDamage(self):
    #     return self.damage
    
    def showUp(self):
        print("name:  ",self.name);
        print("damage:  ",self.damage);
        print("health:  ",self.health);
        
        
        
    
    
        
        

File: Factor/Java/src/factor/Enemy.java
package factor;

public abstract class Enemy {
    private String name;
    private  int damage;
    private  int health;
    public String getName(){
            return name;
    }
    public  void setName(String name){
        this.name = name;
    }
    public int getDamage(){
        return damage;
    }
    public void setDamage(int damage){
        this.damage = damage;
    }
    public int getHealth(){
        return health;
    }
    public void setHealth(int health){
        this.health = health;
    }
    public void showUp(){
        System.out.println("name:  "+name);
        System.out.println("damage:  "+damage);
        System.out.println("health:  "+health);
    }
}
